### ISSUE TITLE
Foreign key support?

### ISSUE THREAD

**Issue by amiel-marqeta** [adangel/liquibase-percona#6]
Hi @adangel,

I've been evaluating this plugin and so far so good. I noticed that when adding a column that has foreign key constraints, pt-osc does add the column however, the FK constraint is not added. 

Sample changeSet

```
<changeSet id="PP-TEST" author="system1" context="master">
    <addColumn tableName="test_table">
        <column name="testColumn" type="bigint(20)">
            <constraints nullable="true"
                         foreignKeyName="fk_test_column"
                         references="test_parent(id)"/>
        </column>
    </addColumn>
</changeSet>
```

Is foreign key creation not supported, if so any reason why?

**Comment by adangel** [comment]
Yes, the constraints are not handled. There is no specific reason, it was simply not needed yet.

What would be the correct way to handle this? I see, that without pt-osc, liquibase would execute an `ALTER TABLE` for adding the column and another `ALTER TABLE` for adding the foreign key constraint. But executing pt-osc twice on the table, doesn't sound fast...

According to https://www.percona.com/doc/percona-toolkit/2.2/pt-online-schema-change.html#cmdoption-pt-online-schema-change--alter it is possible, to add multiple modifications separated by commas. Do you think, this will work?

    --alter "ADD COLUMN testColumn BIGINT(20), ADD CONSTRAINT _fk_test_column FOREIGN KEY (testColumn) REFERENCES test_parent(id)"

There is of course a workaround, by adding a new changeset with [add foreign key constraint change](http://www.liquibase.org/documentation/changes/add_foreign_key_constraint.html). But this would not use pt-osc...  or should it? Then this would be another change that would be executed by liquibase-percona...

What do you think?

**Comment by amiel-marqeta** [comment]
I agree with your assessments. The pt-osc --alter comma delimited command will work and issues only one alter table to add the column and foreign key. I believe adding support for this would be great. This eliminates having pt-osc execute 2 alter tables which is not desired.

In addition ...
```
<addForeignKeyConstraint baseTableName="" baseColumnNames="" constraintName="" referencedTableName=""
                                 referencedColumnNames="" />
```
Should also be executed by liquibase-percona. This use case is when we later decide to add a foreign key after the column has already been added. I believe this is not so common but is nice to have.

**Comment by adangel** [comment]
I've created for the two points two separate issues:

*   #7 which is a bug
*   #8 which is an improvement/feature

**Comment by amiel-marqeta** [comment]
Thanks. I actually worked on these 2 points but sadly I lumped them into one branch with a few changes to the pom.xml since i'm still using an older version of mysql. I could raise a PR if you want.

**Comment by adangel** [comment]
This will be fixed with version 1.3.0 via #7 and #8 .
